plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -912,12 +912,27 @@ public TableStatistics getTableStatistics(ConnectorSession session, JdbcTableHan
     @Override
     public void createSchema(ConnectorSession session, String schemaName)
     {
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            if (connection.getMetaData().storesUpperCaseIdentifiers()) {
+                schemaName = schemaName.toUpperCase(ENGLISH);
+            }
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
         execute(session, "CREATE SCHEMA " + quoted(schemaName));
     }
 
     @Override
     public void dropSchema(ConnectorSession session, String schemaName)
```

Comment:
Please add unit tests when you `CREATE/DROP SCHEMA some_Fancy_schEMA`. 
Also add a test where after `CREATE SCHEMA x`, the `x` schema is created in expected case (depending on `storesUpperCaseIdentifiers()` )

Reply:
Isn't the schema name passed to both `createSchema` and `dropSchema` always lowercase, because Trino is not case sensitive?

Reply:
I added a test BUT it turned out that `toRemoteSchema()` validates that the passed schema name is already lowercase. It was not necessary to call it in `createSchema()`, since we know the schema doesn't exist yet. The test now reflects that. Just please note that this test passes without the changes I made in `createSchema()` and `dropSchema()`.